addons/udes_stock/models/stock_picking_batch.py

```diff
@@ -29,6 +29,38 @@ class StockPickingBatch(models.Model):
         help="Priority of a batch is the maximum priority of its pickings."
     )
 
+    state = fields.Selection(
+        selection_add=[
+            ('waiting', 'Waiting'),
+            ('ready', 'Ready'),
+        ],
+        compute='_compute_state',
+        store=True,
+    )
+
+    @api.multi
+    def confirm_picking(self):
+        """Overwrite method confirm picking to raise error if not in draft and
+           rollback to draft on error in action_assign.
+        """
+        if any(batch.state != 'draft' for batch in self):
+            raise ValidationError(_(
+                'Batch (%s) is not in state draft can not perform '
+                'confirm_picking') % ','.join(
+                    b.name for b in self if b.state != 'draft')
+            )
+
+        pickings_todo = self.mapped('picking_ids')
+        self.write({'state': 'waiting'})  # Get it out of draft
+        try:
+            res = pickings_todo.action_assign()
+        except:
+            # On an error rollback to draft
+            self.write({'state': 'draft'})
+            # Then reraise the error
```

Comment:
Style: the code is clear, I suggest to remove both comments; too much noise otherwise.